app/com/arpnetworking/rollups/RollupGenerator.java

```diff
@@ -51,11 +55,33 @@
 import javax.inject.Named;
 
 /**
- * Actor for performing rollups for individual source metrics.
+ * Actor for generating rollup jobs for individual source metrics.
+ * <p>
+ * The {@code RollupGenerator} will periodically retrieve metrics eligible for rollup and enqueue jobs up
+ * to the maximum number of backfill periods, if any are configured.
  *
  * @author Gilligan Markham (gmarkham at dropbox dot com)
+ *
  */
 public class RollupGenerator extends AbstractActorWithTimers {
+
+    /*
```

Comment:
Not sure where the right place for this is but I would really like some form of in-line documentation explaining the state machine like this. Ideally I would still be able to have it processed by javadoc for formatting, for now I've opted for this unattached comment as I've done here.